pulsar-client/src/main/java/org/apache/pulsar/client/impl/MessageImpl.java

```diff
@@ -270,7 +270,11 @@ public T getValue() {
                     return schema.decode(getData(), schemaVersion);
                 }
             } else {
-                return schema.decode(getData());
+                if (getData().length == 0) {
```

Comment:
@congbobo184 

I think the fix here is a bit problematic. because an application can produce a zero-length bytes array.  Can we find a way to differentiate a zero-length bytes payload from a null payload?

Reply:
Should we store the length of the payload, if we don't store it, we don't know the length of the payload

Reply:
we should use metadata to store the null payload.